File: src/main/java/org/noxet/noxetserver/commands/debug/FakeCombatLog.java
package org.noxet.noxetserver.commands.debug;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.noxet.noxetserver.CombatLogging;
import org.noxet.noxetserver.messaging.ErrorMessage;

public class FakeCombatLog implements CommandExecutor {
    @Override
    @SuppressWarnings("NullableProblems")
    public boolean onCommand(CommandSender commandSender, Command command, String s, String[] strings) {
        if(!commandSender.isOp()) {
            new ErrorMessage(ErrorMessage.ErrorType.PERMISSION, "Only operators can fake combat logging.").send(commandSender);
            return true;
        }

        if(!(commandSender instanceof Player)) {
            new ErrorMessage(ErrorMessage.ErrorType.COMMON, "Only players can fake combat logging.").send(commandSender);
            return true;
        }

        Player player = (Player) commandSender;

        CombatLogging.triggerCombatLog(player);

        return true;
    }
}

File: src/main/java/org/noxet/noxetserver/commands/debug/FakeCombatLog.scala
package org.noxet.noxetserver.commands.debug

import org.bukkit.command.Command
import org.bukkit.entity.Player
import org.noxet.noxetserver.combatlogging.CombatLogging
import org.noxet.noxetserver.commands.RegisteredCommand
import org.noxet.noxetserver.messaging.ErrorMessage

object FakeCombatLog extends OperatorPlayerCommandExecutor, RegisteredCommand("fake-combat-log", this):
  override def onOperatorPlayerCommand(player: Player, command: Command, s: String, strings: Array[String]): Boolean =
    CombatLogging.triggerCombatLog(player)
    true